### PR TITLE
Kyuubi BeeLine wrongly process JDBC URL that contains `--`

### PR DESCRIPTION
**Pull request by zcx513566**
# :mag: Description
## Issue References 🔗

This pull request fixes https://github.com/apache/kyuubi/issues/6522

## Describe Your Solution 🔧

Remove comments in dispatch commands only execute command

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
cc @cfmcgrady @turboFei

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java [hunk 1]
```diff
@@ -282,9 +282,8 @@ int runInit() {
     return executionResult;
   }
 
-  // see HIVE-15820: comment at the head of beeline -e
-  @Override
-  boolean dispatch(String line) {
+  // see HIVE-15820: comment at the head of beeline -e only dispatch commands
+  boolean removeCommentsDispatch(String line) {
```
Comment (turboFei) [inline kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:286]: what isthe difference

Comment (pan3793) [reply kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:286]: The existing implementation also affects invocations in BeeLine
image

**Review by turboFei — APPROVED** [review]

**Comment by turboFei** [comment]
could you add UT?

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java (hunk 1 above)
Comment (zcx513566) [reply kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:286]: > what isthe difference
Remove comments in dispatch commands only execute command. 
for example：beeline -e '--   xxx' or -f script.sql remove comments，but not do anything for other case.

**Comment by zcx513566** [comment]
> could you add UT?
ut file testKyuubiBeelineComment covered remove comments in beeline -e '-- comment' and add ut for url contain '--' case.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merge to master/1.9

**Comment by pan3793** [comment]
@zcx513566 with an additional look at this issue and the [upstream change](https://github.com/apache/hive/commit/425e1ff7c054f87c4db87e77d004282d529599ae), I think the fix is invalid. And the UT you added is invalid too, it also passes without this patch.

**Comment by zcx513566** [comment]
> @zcx513566 with an additional look at this issue and the [upstream change](https://github.com/apache/hive/commit/425e1ff7c054f87c4db87e77d004282d529599ae), I think the fix is invalid. And the UT you added is invalid too, it also passes without this patch.
First the url should not remove comments. it also passes without this patch is  because the UT only check sql content，not verified url.

**Pull request closed without merging**